compiler/daml-lf-tools/src/DA/Daml/LF/Simplifier.hs

```diff
@@ -137,7 +137,7 @@ safetyStep = \case
       BEExpInt64          -> Safe 1
       BEFoldl             -> Safe 2
       BEFoldr             -> Safe 2
-      BEWithAuthorityOf   -> undefined -- TODO https://github.com/digital-asset/daml/issues/15882
+      BEWithAuthority     -> undefined -- TODO https://github.com/digital-asset/daml/issues/15882
```

Comment:
IIUC this should be

```suggestion
      BEWithAuthority     -> Safe 2
```

since it can only crash after being applied to all arguments

Reply:
thanks.
I will make a PR for this.
Do you know what Daml will cause the current `undefined` to be reached?

Reply:
hey @nickchapman-da, sorry I forgot to get back on this. I looked into it and iiuc it looks like it can't be reached because this apples to the _builtin_ `BEWithAuthority` and not to the function `withAuthority`, but in surface daml you can only use the latter.

Reply:
Thanks for this. Let me check I understand...
the `undefined` can't be reached because the primitive will always be fully applied, by construction?

Reply:
sorry, I missed this, but yes that's my reasoning. If you somehow ran the simplifier on handcrafted LF then it would probably be reachable, though.